plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/Coordinates.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven.internal;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import org.sonatype.nexus.common.hash.HashAlgorithm;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Maven non-artifact repository coordinates. Every item in repository may have hashes, stored on paths with proper
+ * suffixes.
```

Comment:
Ack. Will rename it to something like RepositoryPath. The point of the class is to ease interpreting maven repository paths, where /foo or /foo.sha1 or /foo.md5 all actually refers to /foo or some of it's "subordinate" properties like SHA1 or MD5. Also, it simplifies path checking, as removes the need to consider optionally present extensions:

`coordinates.main().getFileName().equals(MAVEN_METADATA_FILENAME)`